lib/src/install.rs

```diff
@@ -368,6 +394,31 @@ impl SourceInfo {
             name: container_info.image.clone(),
         };
         let digest = crate::podman::imageid_to_digest(&container_info.imageid)?;
+
+        let skopeo_supports_containers_storage = skopeo_supports_containers_storage()
+            .context("Failed to run skopeo (it currently must be installed in the host root)")?;
+        Self::from(
+            imageref,
+            digest,
+            Some(HostMountnsInfo {
+                skopeo_supports_containers_storage,
+            }),
+        )
+    }
+
+    #[context("Gathering source info from an external source")]
+    pub(crate) fn from_imageref(imageref: &str) -> Result<Self> {
+        let digest = crate::skopeo::oci_archive_digest(imageref)?;
```

Comment:
Today, the ostree-container stack very much handles just being passed a non-digested pull spec; we'll do the digest resolution there. An alternative here is to make `digest` an `Option<T>` instead.  I have a mild preference for this, but I'm also fine with the code as is.

Reply:
I hope I understood your comment correctly. I made `digest` an `Option<T>`. This loses a bit of "null-safety" (handled gracefully, ofc), but it allowed me to drop the whole `skopeo` crate. I have a backup of the old code if we change our midns.